src/apiGateway.ts

```diff
@@ -0,0 +1,128 @@
+
+import * as aws from "@pulumi/aws";
+
+import { EnforcementLevel, ResourceValidationPolicy, validateTypedResource } from "@pulumi/policy";
+
+import { registerPolicy } from "./awsGuard";
+import { defaultEnforcementLevel } from "./enforcementLevel";
+import { PolicyArgs } from "./policyArgs";
+import { getValueOrDefault } from "./util";
+
+// Mixin additional properties onto AwsGuardArgs.
+declare module "./awsGuard" {
+    interface AwsGuardArgs {
+        apiGatewayStageCached?: EnforcementLevel;
+        apiGatewayMethodCachedAndEncrypted?: EnforcementLevel;
+        apiGatewayEndpointTypeCheck?: EnforcementLevel | ApiGatewayEndpointTypeCheckArgs;
+    }
+}
+
+// Register policy factories.
+registerPolicy("apiGatewayStageCached", apiGatewayStageCached);
+registerPolicy("apiGatewayMethodCachedAndEncrypted", apiGatewayMethodCachedAndEncrypted);
+registerPolicy("apiGatewayEndpointTypeCheck", apiGatewayEndpointTypeCheck);
+
+/** @internal */
+export function apiGatewayStageCached(enforcementLevel?: EnforcementLevel): ResourceValidationPolicy {
+    return {
+        name: "apigateway-stage-cached",
+        description: "Checks that API Gateway Stages have a cache cluster enabled.",
+        enforcementLevel: enforcementLevel || defaultEnforcementLevel,
+        validateResource: validateTypedResource(aws.apigateway.Stage, (stage, _, reportViolation) => {
+            if (!stage.cacheClusterEnabled) {
+                reportViolation(`API Gateway Stage '${stage.stageName}' does not have a cache cluster enabled.`);
+            }
+        }),
+    };
+}
+
+/** @internal */
+export function apiGatewayMethodCachedAndEncrypted(enforcementLevel?: EnforcementLevel): ResourceValidationPolicy {
+    return {
+        name: "apigateway-method-cached-and-encrypted",
+        description: "Checks API Gateway Methods that responses are configured to be cached and that those cached responses are encrypted.",
+        enforcementLevel: enforcementLevel || defaultEnforcementLevel,
+        validateResource: validateTypedResource(aws.apigateway.MethodSettings, (methodSettings, _, reportViolation) => {
+            if (!methodSettings.settings.cachingEnabled) {
+                reportViolation(`API Gateway Method '${methodSettings.methodPath}' does not have caching enabled.`);
+            }
+            if (!methodSettings.settings.cacheDataEncrypted) {
+                reportViolation(`API Gateway Method '${methodSettings.methodPath}' not configured to encrypt cached responses.`);
+            }
+        }),
+    };
+}
+
+export interface ApiGatewayEndpointTypeCheckArgs extends PolicyArgs {
+    /** Whether or not API Endpoint type EDGE is allowed. */
+    allowEdge?: boolean;
+    /** Whether or not API Endpoint type REGIONAL is allowed. */
+    allowRegional?: boolean;
+    /** Whether or not API Endpoint type PRIVATE is allowed. */
+    allowPrivate?: boolean;
+}
+
+/** @internal */
+export function apiGatewayEndpointTypeCheck(
```

Comment:
Should we get rid of the "check" here and below? I believe thats what we did for others.

Reply:
Yeah, looks like it. Done.